src/transformers/models/mamba/modeling_mamba.py

```diff
@@ -224,16 +225,16 @@ def forward(self, hidden_states, inference_params=None):
 
         else:
             inference_params.conv_states[self.layer_idx].copy_(
-                nn.functional.pad(hidden_states, (self.d_conv - hidden_states.shape[-1], 0))
+                nn.functional.pad(hidden_states, (self.conv_kernel_size - hidden_states.shape[-1], 0))
```

Comment:
is there something wrong, because may the pad_token is not zero. Or there is no ref the pad_token.

Reply:
it's not related to the padding token, we are padding hidden states with just zero. There is not attention so no padding mask / no attention mask. If we don't want the padding to be taken into account the padding embedding should return zeros for pads

Reply:
is the model support the original trainer? I tried this but get the error

![image](https://github.com/huggingface/transformers/assets/53668275/bd6219f8-518e-438f-9e29-d33a8d742cea)

if is there should inherit the trainer class and modify something?

```
from transformers import Trainer, TrainingArguments

training_args = TrainingArguments(
    output_dir="./mamba",
    overwrite_output_dir=True,
    num_train_epochs=1,
    per_device_train_batch_size=1,
    save_steps=10_000,
    save_total_limit=2,
)

trainer = Trainer(
    model=model,
    args=training_args,
    train_dataset=data
)

trainer.train()
```